internal/sinks/sink.go

```diff
@@ -0,0 +1,37 @@
+package sinks
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/go-logr/logr"
+
+	"github.com/spectrocloud-labs/validator/api/v1alpha1"
+)
+
+type Sink interface {
+	Configure(c Client, vc v1alpha1.ValidatorConfig, config map[string][]byte) error
+	Emit(result v1alpha1.ValidationResult) error
+}
+
+func NewSink(sinkType string, log logr.Logger) Sink {
+	switch sinkType {
+	case "slack":
+		return &SlackSink{log: log}
+	default:
+		return &SlackSink{log: log}
```

Comment:
Should this default to a "NilSink"? ie a sink that has Configure and Emit functions which just return nil